src/theory/strings/array_core_solver.cpp

```diff
@@ -95,63 +95,68 @@ void ArrayCoreSolver::checkUpdate(const std::vector<Node>& updateTerms)
     // inference rule is:
     // (seq.update x i a) in TERMS
     // (seq.nth t j) in TERMS
-    // t == (seq.update x i a)
-    // ----------------------------------------------------------------------
+    // t = (seq.update x i a)
+    // ---------------------------------------------------------------------
     // (seq.nth (seq.update x i a) j) =
     //   (ITE, j in range(i, i+len(a)), (seq.nth a (j - i)),  (seq.nth x j))
 
-    // t == (seq.update x i a) =>
-    // (seq.nth t j) = (ITE, j in range(i, i+len(a)), (seq.nth a (j - i)),
-    // (seq.nth x j))
-
     // note that the term could rewrites to a skolem
     // get proxy variable for the update term as t
     Node termProxy = d_termReg.getProxyVariableFor(n);
     Trace("seq-update") << "- " << termProxy << " = " << n << std::endl;
     std::vector<Node> exp;
     d_im.addToExplanation(termProxy, n, exp);
 
-    // optimization: add a short cut t == (seq.update n[0] n[1] n[2]) => t[i] ==
-    // n[2][0]
+    // optimization: add a short cut
+    // t = (seq.update n[0] n[1] n[2])
+    // ---------------------------------------------------------------------
+    // nth(t, n[1]) = (ITE, len(n[0]) > 0, nth(n[2], 0), nth(n[0], n[1]))
```

Comment:
I don't think that this lemma is correct, because the length of `n[0]` should not matter. Rather, what matters is that `n[1]` is within `0 <= n[1] < len(n[0])`. Am I missing something?

Reply:
You are right. Have been updated.